### PR TITLE
add more information when `service.platform` isn't part of `service.build.platforms`

### PR DESCRIPTION
**Pull request by bkielbasa**
**What I did**

I found it a bit problematic to debug so I added a bit more detailed error messages when `service.platform` isn't part of `service.build.platforms`.

### CODE REVIEW

**Comment by glours** [comment]
@bkielbasa can you sign-off your commit? Just follow the instructions [here](https://github.com/docker/compose/pull/9868/checks?check_run_id=8510699917)

File: pkg/compose/build.go [hunk 1]
```diff
@@ -370,7 +370,7 @@ func addPlatforms(project *types.Project, service types.ServiceConfig) ([]specs.
 
 	if service.Platform != "" && !utils.StringContains(service.Build.Platforms, service.Platform) {
 		if len(service.Build.Platforms) > 0 {
-			return nil, fmt.Errorf("service.platform should be part of the service.build.platforms: %q", service.Platform)
+			return nil, fmt.Errorf("service.platform %q should be part of the service.build.platforms: %q", service.Platform, service.Build.Platforms)
```
Comment (glours) [inline pkg/compose/build.go:373]: To be coherent, we should also update the error log line 355, can you apply the same modification?

You also need to update `TestBuildPlatformsStandardErrors` end-to-end tests in the [build_test.go ](https://github.com/docker/compose/blob/v2/pkg/e2e/build_test.go) file

**Comment by bkielbasa** [comment]
@glours thanks for your review! I updated tests so it should be ready to go!

**Review by glours — CHANGES_REQUESTED** [review]
As your branch isn't up to date with the v2 branch, we can't merge your PR or we'll introduce potential regression. 
The only changes we should see in this PR are:
* The 2 updates of the logs
* The 2 updates in the e2e test assertions

**Comment by bkielbasa** [comment]
@glours done! Sorry, I mess up with rebase :) it should be fine now

**Review by glours — APPROVED** [review]
LGTM, thanks for your contribution

**Pull request merged**